src/PinchZoomPan.js

```diff
@@ -166,8 +181,29 @@ export default class PinchZoomPan extends React.Component {
         this.hasMovedImg = false;
     };
 
+    handleDragEnd = () => {
+        this.dragCoords.endTime = Date.now();
+        const { onDragEnd } = this.props;
+        const { startX, endX, startY, endY, startTime, endTime, isMoved } = this.dragCoords;
+        const deltaX = endX - startX;
+        const deltaY = endY - startY;
+
+        const distance = Math.sqrt(deltaX * deltaX + deltaY * deltaY);
+        const timeDiff = endTime - startTime;
+        const velocity = distance / timeDiff;
+
+        let horizontalDragDirection = deltaX > 0 ? 'right' : 'left';
+        let verticalDragDirection = deltaY > 0 ? 'down' : 'up';
```

Comment:
Do we ever need vertical direction, velocity ?

Reply:
In the future, we might need.